### PR TITLE
Fix error when using `kubectl autoscale` command with dryrun flag

### PR DESCRIPTION
**Pull request by xingzhou**
While working on #39833, I found the following issue:
When using `dry-run` flag in `kubectl autoscale` command, it
always print error message "no object passed to autoscale" at
the end of the output.

This patch fixed this error and refined the output message when
using `dry-run` flag.

### CODE REVIEW

**Comment by grodrigues3** [comment]
@k8s-bot ok to test

**Comment by xingzhou** [comment]
@k8s-bot bazel test this

**Comment by deads2k** [comment]
@kubernetes/sig-cli-pr-reviews

File: pkg/kubectl/cmd/autoscale.go [hunk 1]
```diff
@@ -162,7 +161,11 @@ func RunAutoscale(f cmdutil.Factory, out io.Writer, cmd *cobra.Command, args []s
 			object = hpa.Object
 		}
 		if cmdutil.GetDryRunFlag(cmd) {
-			return f.PrintObject(cmd, mapper, object, out)
+			if len(cmdutil.GetFlagString(cmd, "output")) > 0 {
+				return f.PrintObject(cmd, mapper, object, out)
+			}
+			cmdutil.PrintSuccess(mapper, false, out, info.Mapping.Resource, info.Name, true, "autoscaled")
```
Comment (adohe) [inline pkg/kubectl/cmd/autoscale.go]: this overall looks good to me. But I would like change this to be something more accurate, under dry-run, the object will not be autoscaled.

**Comment by adohe** [comment]
@xingzhou ptal. I will apply label once comment addressed.

File: pkg/kubectl/cmd/autoscale.go (hunk 1 above)
Comment (xingzhou) [inline pkg/kubectl/cmd/autoscale.go]: @adohe, according to other command's dry-run flag, this command will print sth like:
"deployment "dep" autoscaled (dry run)"
don't you think the tail `dry run` message is not clear enough?

**Pull request closed without merging**